insights/specs/datasources/aws.py

```diff
@@ -0,0 +1,42 @@
+"""
+Custom datasources for aws information
+"""
+from insights.components.cloud_provider import IsAWS
+from insights.core.context import HostContext
+from insights.core.dr import SkipComponent
+from insights.core.plugins import datasource
+from insights.core.spec_factory import simple_command
+from insights.specs import Specs
+
+
+class LocalSpecs(Specs):
+    """ Local specs used only by aws datasources """
+    aws_imdsv2_token = simple_command(
+        '/usr/bin/curl -s -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 21600" --connect-timeout 5',
```

Comment:
This update looks good to me and per my test, it works well on AWS instances.

However, my only concern is, that the raw token will be present in the collected file name. I'm not sure if this would increase the attack surface to the customer's EC2 instances, but it sounds like a good idea to limit the token period to a shorter valid period, e.g. `60` seconds, what do you think?

Reply:
@xiangce that is a good suggestion, thank you.  I'll make the change.